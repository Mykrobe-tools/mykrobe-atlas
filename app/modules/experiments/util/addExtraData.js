/* @flow */

export default (json: Object) => {
  const testData = require('../__fixtures__/experiment.json').data;
  if (!json.location) {
    json.location = testData.location;
  }
  if (!json.geoDistance) {
    json.geoDistance = testData.geoDistance;
  }
  if (json.geoDistance.experiments.length === 0) {
    json.geoDistance.experiments = testData.geoDistance.experiments;
  }
  if (!json.snpDistance) {
    json.snpDistance = testData.snpDistance;
  }
  if (!json.snpDistance.newick) {
    json.snpDistance.newick = testData.snpDistance.newick;
  }
  return json;
};
