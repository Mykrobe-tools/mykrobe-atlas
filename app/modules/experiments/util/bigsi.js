/* @flow */

import _get from 'lodash.get';

export const mapTypeToSearch = {
  sequence: 'Sequence search',
  'protein-variant': 'Protein variant search',
  'dna-variant': 'DNA variant search',
};

export const descriptionForBigsi = (bigsi) => {
  const query = _get(bigsi, 'query.seq');
  const type = _get(bigsi, 'type');
  const search = mapTypeToSearch[type] || 'Search';
  return query ? `${search} for ${query}` : search;
};
