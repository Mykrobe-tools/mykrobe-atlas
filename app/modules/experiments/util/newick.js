/* @flow */

import memoizeOne from 'memoize-one';

export const getNewickTokens = memoizeOne(newick =>
  newick.split(/\s*(;|\(|\)|,|:)\s*/)
);

export const newickContainsNodeId = (newick, nodeId) => {
  const tokens = getNewickTokens(newick);
  return tokens && tokens.includes(nodeId);
};
