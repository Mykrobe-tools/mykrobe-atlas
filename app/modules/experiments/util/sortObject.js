/* @flow */

const sortObject = (o: Object) => {
  let sorted = {};
  let key;
  let a = [];

  for (key in o) {
    if (o.hasOwnProperty(key)) {
      a.push(key);
    }
  }

  a.sort();

  for (key = 0; key < a.length; key++) {
    sorted[a[key]] = o[a[key]];
  }
  return sorted;
};

export default sortObject;
