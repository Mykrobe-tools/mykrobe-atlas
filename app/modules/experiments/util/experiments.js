/* @flow */

import _get from 'lodash.get';

import { newickContainsNodeId } from './newick';

export const experimentsInTree = (
  experimentsTree,
  experiments,
  inTree = true
) => {
  console.log('experimentsInTree', { experimentsTree, experiments });
  return (
    experiments &&
    experiments.filter((experiment) => {
      const leafId = experiment?.leafId;
      let contains = false;
      if (leafId) {
        contains = newickContainsNodeId(experimentsTree, leafId);
      }
      if (contains) {
        return inTree;
      }
      return !inTree;
    })
  );
};

export const experimentsWithGeolocation = (
  experiments,
  withGeolocation = true
) => {
  return (
    experiments &&
    experiments.filter((experiment) => {
      const longitudeIsolate = _get(
        experiment,
        'metadata.sample.longitudeIsolate'
      );
      const latitudeIsolate = _get(
        experiment,
        'metadata.sample.latitudeIsolate'
      );
      if (longitudeIsolate && latitudeIsolate) {
        return withGeolocation;
      }
      return !withGeolocation;
    })
  );
};
