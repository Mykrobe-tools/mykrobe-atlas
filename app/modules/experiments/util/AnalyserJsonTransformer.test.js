/* @flow */

import fs from 'fs';
import path from 'path';

import { EXEMPLAR_SEQUENCE_DATA_ARTEFACT_JSON_FOLDER_PATH } from '../../../../desktop/util';

import AnalyserJsonTransformer from './AnalyserJsonTransformer';
import susceptibilityTransformer from './transformers/susceptibility';

const samples = [
  'INH_monoresistant.fastq.gz.json',
  'MDR.fastq.gz.json',
  'Mycobacterium_kansasii.bam.json',
  'RIF_monoresistant.fastq.gz.json',
  'XDR.fastq.gz.json',
];

describe('AnalyserJsonTransformer', () => {
  it('should transform as expected', async () => {
    const transformer = new AnalyserJsonTransformer();
    for (let i = 0; i < samples.length; i++) {
      const sample = samples[i];
      const raw = fs.readFileSync(
        path.join(EXEMPLAR_SEQUENCE_DATA_ARTEFACT_JSON_FOLDER_PATH, sample),
        'utf8'
      );
      const json = JSON.parse(raw);
      const transformed = await transformer.transformModel(json);
      expect(transformed).toMatchSnapshot();
    }
  });
});

describe('AnalyserJsonTransformer susceptibilityTransformer', () => {
  it('should transform as expected', async () => {
    for (let i = 0; i < samples.length; i++) {
      const sample = samples[i];
      const raw = fs.readFileSync(
        path.join(EXEMPLAR_SEQUENCE_DATA_ARTEFACT_JSON_FOLDER_PATH, sample),
        'utf8'
      );
      const json = JSON.parse(raw);
      const keys = Object.keys(json);
      const first = json[keys[0]];
      const susceptibility = first['susceptibility'];
      const transformed = susceptibilityTransformer(susceptibility);
      expect(transformed).toMatchSnapshot();
    }
  });
});
