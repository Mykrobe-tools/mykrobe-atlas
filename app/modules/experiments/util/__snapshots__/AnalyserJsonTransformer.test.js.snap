// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnalyserJsonTransformer should transform as expected 1`] = `
Object {
  "drugsResistance": Object {
    "mdr": false,
    "xdr": false,
  },
  "evidence": Object {
    "Isoniazid": Array [
      Array [
        "Resistance mutation found: S315X in gene katG",
        "1633 kmers from the resistant allele",
        "14 kmers from the susceptible allele",
      ],
    ],
  },
  "hasResistance": true,
  "hasSpecies": true,
  "inconclusive": Array [],
  "inducible": Array [],
  "lineage": Array [
    "European_American",
  ],
  "lineageString": "European American",
  "negative": Array [],
  "positive": Array [],
  "resistant": Array [
    "Isoniazid",
  ],
  "species": Array [
    "Mycobacterium_tuberculosis",
  ],
  "speciesAndLineageString": "Mycobacterium tuberculosis (lineage: European American)",
  "speciesString": "Mycobacterium tuberculosis",
  "susceptibility": Object {
    "Amikacin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Capreomycin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Ciprofloxacin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Ethambutol": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Isoniazid": Object {
      "inconclusive": false,
      "inducible": false,
      "mutation": "katG (S315X)",
      "resistant": true,
      "susceptible": false,
    },
    "Kanamycin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Moxifloxacin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Ofloxacin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Pyrazinamide": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Rifampicin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Streptomycin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
  },
  "susceptible": Array [
    "Ofloxacin",
    "Moxifloxacin",
    "Kanamycin",
    "Ethambutol",
    "Streptomycin",
    "Ciprofloxacin",
    "Pyrazinamide",
    "Rifampicin",
    "Amikacin",
    "Capreomycin",
  ],
}
`;

exports[`AnalyserJsonTransformer should transform as expected 2`] = `
Object {
  "drugsResistance": Object {
    "mdr": true,
    "xdr": false,
  },
  "evidence": Object {
    "Isoniazid": Array [
      Array [
        "Resistance mutation found: S315X in gene katG",
        "850 kmers from the resistant allele",
        "0 kmers from the susceptible allele",
      ],
    ],
    "Rifampicin": Array [
      Array [
        "Resistance mutation found: S450X in gene rpoB",
        "960 kmers from the resistant allele",
        "0 kmers from the susceptible allele",
      ],
    ],
  },
  "hasResistance": true,
  "hasSpecies": true,
  "inconclusive": Array [],
  "inducible": Array [],
  "lineage": Array [
    "European_American",
  ],
  "lineageString": "European American",
  "negative": Array [],
  "positive": Array [],
  "resistant": Array [
    "Isoniazid",
    "Rifampicin",
  ],
  "species": Array [
    "Mycobacterium_tuberculosis",
  ],
  "speciesAndLineageString": "Mycobacterium tuberculosis (lineage: European American)",
  "speciesString": "Mycobacterium tuberculosis",
  "susceptibility": Object {
    "Amikacin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Capreomycin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Ciprofloxacin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Ethambutol": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Isoniazid": Object {
      "inconclusive": false,
      "inducible": false,
      "mutation": "katG (S315X)",
      "resistant": true,
      "susceptible": false,
    },
    "Kanamycin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Moxifloxacin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Ofloxacin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Pyrazinamide": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Rifampicin": Object {
      "inconclusive": false,
      "inducible": false,
      "mutation": "rpoB (S450X)",
      "resistant": true,
      "susceptible": false,
    },
    "Streptomycin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
  },
  "susceptible": Array [
    "Ofloxacin",
    "Moxifloxacin",
    "Kanamycin",
    "Ethambutol",
    "Streptomycin",
    "Ciprofloxacin",
    "Pyrazinamide",
    "Amikacin",
    "Capreomycin",
  ],
}
`;

exports[`AnalyserJsonTransformer should transform as expected 3`] = `
Object {
  "hasResistance": false,
  "hasSpecies": true,
  "lineage": Array [
    "Unknown",
  ],
  "lineageString": "Unknown",
  "species": Array [
    "Mycobacterium_kansasii",
  ],
  "speciesAndLineageString": "Mycobacterium kansasii (lineage: Unknown)",
  "speciesString": "Mycobacterium kansasii",
}
`;

exports[`AnalyserJsonTransformer should transform as expected 4`] = `
Object {
  "drugsResistance": Object {
    "mdr": false,
    "xdr": false,
  },
  "evidence": Object {
    "Rifampicin": Array [
      Array [
        "Resistance mutation found: H445X in gene rpoB",
        "558 kmers from the resistant allele",
        "2016 kmers from the susceptible allele",
      ],
      Array [
        "Resistance mutation found: S450X in gene rpoB",
        "1922 kmers from the resistant allele",
        "745 kmers from the susceptible allele",
      ],
    ],
  },
  "hasResistance": true,
  "hasSpecies": true,
  "inconclusive": Array [],
  "inducible": Array [],
  "lineage": Array [
    "European_American",
  ],
  "lineageString": "European American",
  "negative": Array [],
  "positive": Array [],
  "resistant": Array [
    "Rifampicin",
  ],
  "species": Array [
    "Mycobacterium_tuberculosis",
  ],
  "speciesAndLineageString": "Mycobacterium tuberculosis (lineage: European American)",
  "speciesString": "Mycobacterium tuberculosis",
  "susceptibility": Object {
    "Amikacin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Capreomycin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Ciprofloxacin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Ethambutol": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Isoniazid": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Kanamycin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Moxifloxacin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Ofloxacin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Pyrazinamide": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Rifampicin": Object {
      "inconclusive": false,
      "inducible": false,
      "mutation": "rpoB (S450X)",
      "resistant": true,
      "susceptible": false,
    },
    "Streptomycin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
  },
  "susceptible": Array [
    "Ofloxacin",
    "Moxifloxacin",
    "Isoniazid",
    "Kanamycin",
    "Ethambutol",
    "Streptomycin",
    "Ciprofloxacin",
    "Pyrazinamide",
    "Amikacin",
    "Capreomycin",
  ],
}
`;

exports[`AnalyserJsonTransformer should transform as expected 5`] = `
Object {
  "drugsResistance": Object {
    "mdr": true,
    "xdr": true,
  },
  "evidence": Object {
    "Amikacin": Array [
      Array [
        "Resistance mutation found: A1401X in gene rrs",
        "1025 kmers from the resistant allele",
        "0 kmers from the susceptible allele",
      ],
    ],
    "Capreomycin": Array [
      Array [
        "Resistance mutation found: A1401X in gene rrs",
        "1025 kmers from the resistant allele",
        "0 kmers from the susceptible allele",
      ],
    ],
    "Ciprofloxacin": Array [
      Array [
        "Resistance mutation found: A90V in gene gyrA",
        "252 kmers from the resistant allele",
        "432 kmers from the susceptible allele",
      ],
      Array [
        "Resistance mutation found: D94G in gene gyrA",
        "360 kmers from the resistant allele",
        "227 kmers from the susceptible allele",
      ],
    ],
    "Ethambutol": Array [
      Array [
        "Resistance mutation found: M306V in gene embB",
        "871 kmers from the resistant allele",
        "0 kmers from the susceptible allele",
      ],
    ],
    "Isoniazid": Array [
      Array [
        "Resistance mutation found: S315X in gene katG",
        "606 kmers from the resistant allele",
        "0 kmers from the susceptible allele",
      ],
    ],
    "Kanamycin": Array [
      Array [
        "Resistance mutation found: A1401X in gene rrs",
        "1025 kmers from the resistant allele",
        "0 kmers from the susceptible allele",
      ],
    ],
    "Moxifloxacin": Array [
      Array [
        "Resistance mutation found: A90V in gene gyrA",
        "252 kmers from the resistant allele",
        "432 kmers from the susceptible allele",
      ],
      Array [
        "Resistance mutation found: D94G in gene gyrA",
        "360 kmers from the resistant allele",
        "227 kmers from the susceptible allele",
      ],
    ],
    "Ofloxacin": Array [
      Array [
        "Resistance mutation found: A90V in gene gyrA",
        "252 kmers from the resistant allele",
        "432 kmers from the susceptible allele",
      ],
      Array [
        "Resistance mutation found: D94G in gene gyrA",
        "360 kmers from the resistant allele",
        "227 kmers from the susceptible allele",
      ],
    ],
    "Pyrazinamide": Array [
      Array [
        "Resistance mutation found: H137R in gene pncA",
        "123 kmers from the resistant allele",
        "446 kmers from the susceptible allele",
      ],
    ],
    "Rifampicin": Array [
      Array [
        "Resistance mutation found: S450X in gene rpoB",
        "844 kmers from the resistant allele",
        "0 kmers from the susceptible allele",
      ],
    ],
    "Streptomycin": Array [
      Array [
        "Resistance mutation found: K43R in gene rpsL",
        "1060 kmers from the resistant allele",
        "0 kmers from the susceptible allele",
      ],
    ],
  },
  "hasResistance": true,
  "hasSpecies": true,
  "inconclusive": Array [],
  "inducible": Array [],
  "lineage": Array [
    "Beijing_East_Asia",
  ],
  "lineageString": "Beijing East Asia",
  "negative": Array [],
  "positive": Array [],
  "resistant": Array [
    "Ofloxacin",
    "Moxifloxacin",
    "Isoniazid",
    "Kanamycin",
    "Ethambutol",
    "Streptomycin",
    "Ciprofloxacin",
    "Pyrazinamide",
    "Rifampicin",
    "Amikacin",
    "Capreomycin",
  ],
  "species": Array [
    "Mycobacterium_tuberculosis",
  ],
  "speciesAndLineageString": "Mycobacterium tuberculosis (lineage: Beijing East Asia)",
  "speciesString": "Mycobacterium tuberculosis",
  "susceptibility": Object {
    "Amikacin": Object {
      "inconclusive": false,
      "inducible": false,
      "mutation": "rrs (A1401X)",
      "resistant": true,
      "susceptible": false,
    },
    "Capreomycin": Object {
      "inconclusive": false,
      "inducible": false,
      "mutation": "rrs (A1401X)",
      "resistant": true,
      "susceptible": false,
    },
    "Ciprofloxacin": Object {
      "inconclusive": false,
      "inducible": false,
      "mutation": "gyrA (D94G)",
      "resistant": true,
      "susceptible": false,
    },
    "Ethambutol": Object {
      "inconclusive": false,
      "inducible": false,
      "mutation": "embB (M306V)",
      "resistant": true,
      "susceptible": false,
    },
    "Isoniazid": Object {
      "inconclusive": false,
      "inducible": false,
      "mutation": "katG (S315X)",
      "resistant": true,
      "susceptible": false,
    },
    "Kanamycin": Object {
      "inconclusive": false,
      "inducible": false,
      "mutation": "rrs (A1401X)",
      "resistant": true,
      "susceptible": false,
    },
    "Moxifloxacin": Object {
      "inconclusive": false,
      "inducible": false,
      "mutation": "gyrA (D94G)",
      "resistant": true,
      "susceptible": false,
    },
    "Ofloxacin": Object {
      "inconclusive": false,
      "inducible": false,
      "mutation": "gyrA (D94G)",
      "resistant": true,
      "susceptible": false,
    },
    "Pyrazinamide": Object {
      "inconclusive": false,
      "inducible": false,
      "mutation": "pncA (H137R)",
      "resistant": true,
      "susceptible": false,
    },
    "Rifampicin": Object {
      "inconclusive": false,
      "inducible": false,
      "mutation": "rpoB (S450X)",
      "resistant": true,
      "susceptible": false,
    },
    "Streptomycin": Object {
      "inconclusive": false,
      "inducible": false,
      "mutation": "rpsL (K43R)",
      "resistant": true,
      "susceptible": false,
    },
  },
  "susceptible": Array [],
}
`;

exports[`AnalyserJsonTransformer susceptibilityTransformer should transform as expected 1`] = `
Object {
  "evidence": Object {
    "Isoniazid": Array [
      Array [
        "Resistance mutation found: S315X in gene katG",
        "Resistant allele coverage: 82",
        "Susceptible allele coverage: 1",
      ],
    ],
  },
  "inconclusive": Array [],
  "inducible": Array [],
  "resistant": Array [
    "Isoniazid",
  ],
  "susceptibility": Object {
    "Amikacin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Capreomycin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Ciprofloxacin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Ethambutol": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Isoniazid": Object {
      "inconclusive": false,
      "inducible": false,
      "mutation": "katG (S315X)",
      "resistant": true,
      "susceptible": false,
    },
    "Kanamycin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Moxifloxacin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Ofloxacin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Pyrazinamide": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Rifampicin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Streptomycin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
  },
  "susceptible": Array [
    "Ofloxacin",
    "Moxifloxacin",
    "Kanamycin",
    "Ethambutol",
    "Streptomycin",
    "Ciprofloxacin",
    "Pyrazinamide",
    "Rifampicin",
    "Amikacin",
    "Capreomycin",
  ],
}
`;

exports[`AnalyserJsonTransformer susceptibilityTransformer should transform as expected 2`] = `
Object {
  "evidence": Object {
    "Isoniazid": Array [
      Array [
        "Resistance mutation found: S315X in gene katG",
        "Resistant allele coverage: 43",
        "Susceptible allele coverage: 0",
      ],
    ],
    "Rifampicin": Array [
      Array [
        "Resistance mutation found: S450X in gene rpoB",
        "Resistant allele coverage: 51",
        "Susceptible allele coverage: 0",
      ],
    ],
  },
  "inconclusive": Array [],
  "inducible": Array [],
  "resistant": Array [
    "Isoniazid",
    "Rifampicin",
  ],
  "susceptibility": Object {
    "Amikacin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Capreomycin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Ciprofloxacin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Ethambutol": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Isoniazid": Object {
      "inconclusive": false,
      "inducible": false,
      "mutation": "katG (S315X)",
      "resistant": true,
      "susceptible": false,
    },
    "Kanamycin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Moxifloxacin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Ofloxacin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Pyrazinamide": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Rifampicin": Object {
      "inconclusive": false,
      "inducible": false,
      "mutation": "rpoB (S450X)",
      "resistant": true,
      "susceptible": false,
    },
    "Streptomycin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
  },
  "susceptible": Array [
    "Ofloxacin",
    "Moxifloxacin",
    "Kanamycin",
    "Ethambutol",
    "Streptomycin",
    "Ciprofloxacin",
    "Pyrazinamide",
    "Amikacin",
    "Capreomycin",
  ],
}
`;

exports[`AnalyserJsonTransformer susceptibilityTransformer should transform as expected 3`] = `
Object {
  "evidence": Object {},
  "inconclusive": Array [],
  "inducible": Array [],
  "resistant": Array [],
  "susceptibility": Object {
    "Amikacin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Capreomycin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Ciprofloxacin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Ethambutol": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Isoniazid": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Kanamycin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Moxifloxacin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Ofloxacin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Pyrazinamide": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Rifampicin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Streptomycin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
  },
  "susceptible": Array [
    "Ofloxacin",
    "Moxifloxacin",
    "Isoniazid",
    "Kanamycin",
    "Ethambutol",
    "Streptomycin",
    "Ciprofloxacin",
    "Pyrazinamide",
    "Rifampicin",
    "Amikacin",
    "Capreomycin",
  ],
}
`;

exports[`AnalyserJsonTransformer susceptibilityTransformer should transform as expected 4`] = `
Object {
  "evidence": Object {
    "Rifampicin": Array [
      Array [
        "Resistance mutation found: H445X in gene rpoB",
        "Resistant allele coverage: 28",
        "Susceptible allele coverage: 98",
      ],
      Array [
        "Resistance mutation found: S450X in gene rpoB",
        "Resistant allele coverage: 100",
        "Susceptible allele coverage: 40",
      ],
    ],
  },
  "inconclusive": Array [],
  "inducible": Array [],
  "resistant": Array [
    "Rifampicin",
  ],
  "susceptibility": Object {
    "Amikacin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Capreomycin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Ciprofloxacin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Ethambutol": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Isoniazid": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Kanamycin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Moxifloxacin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Ofloxacin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Pyrazinamide": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
    "Rifampicin": Object {
      "inconclusive": false,
      "inducible": false,
      "mutation": "rpoB (S450X)",
      "resistant": true,
      "susceptible": false,
    },
    "Streptomycin": Object {
      "inconclusive": false,
      "inducible": false,
      "resistant": false,
      "susceptible": true,
    },
  },
  "susceptible": Array [
    "Ofloxacin",
    "Moxifloxacin",
    "Isoniazid",
    "Kanamycin",
    "Ethambutol",
    "Streptomycin",
    "Ciprofloxacin",
    "Pyrazinamide",
    "Amikacin",
    "Capreomycin",
  ],
}
`;

exports[`AnalyserJsonTransformer susceptibilityTransformer should transform as expected 5`] = `
Object {
  "evidence": Object {
    "Amikacin": Array [
      Array [
        "Resistance mutation found: A1401X in gene rrs",
        "Resistant allele coverage: 52",
        "Susceptible allele coverage: 0",
      ],
    ],
    "Capreomycin": Array [
      Array [
        "Resistance mutation found: A1401X in gene rrs",
        "Resistant allele coverage: 52",
        "Susceptible allele coverage: 0",
      ],
    ],
    "Ciprofloxacin": Array [
      Array [
        "Resistance mutation found: A90V in gene gyrA",
        "Resistant allele coverage: 13",
        "Susceptible allele coverage: 22",
      ],
      Array [
        "Resistance mutation found: D94G in gene gyrA",
        "Resistant allele coverage: 18",
        "Susceptible allele coverage: 11",
      ],
    ],
    "Ethambutol": Array [
      Array [
        "Resistance mutation found: M306V in gene embB",
        "Resistant allele coverage: 45",
        "Susceptible allele coverage: 0",
      ],
    ],
    "Isoniazid": Array [
      Array [
        "Resistance mutation found: S315X in gene katG",
        "Resistant allele coverage: 30",
        "Susceptible allele coverage: 0",
      ],
    ],
    "Kanamycin": Array [
      Array [
        "Resistance mutation found: A1401X in gene rrs",
        "Resistant allele coverage: 52",
        "Susceptible allele coverage: 0",
      ],
    ],
    "Moxifloxacin": Array [
      Array [
        "Resistance mutation found: A90V in gene gyrA",
        "Resistant allele coverage: 13",
        "Susceptible allele coverage: 22",
      ],
      Array [
        "Resistance mutation found: D94G in gene gyrA",
        "Resistant allele coverage: 18",
        "Susceptible allele coverage: 11",
      ],
    ],
    "Ofloxacin": Array [
      Array [
        "Resistance mutation found: A90V in gene gyrA",
        "Resistant allele coverage: 13",
        "Susceptible allele coverage: 22",
      ],
      Array [
        "Resistance mutation found: D94G in gene gyrA",
        "Resistant allele coverage: 18",
        "Susceptible allele coverage: 11",
      ],
    ],
    "Pyrazinamide": Array [
      Array [
        "Resistance mutation found: H137R in gene pncA",
        "Resistant allele coverage: 7",
        "Susceptible allele coverage: 22",
      ],
    ],
    "Rifampicin": Array [
      Array [
        "Resistance mutation found: S450X in gene rpoB",
        "Resistant allele coverage: 44",
        "Susceptible allele coverage: 0",
      ],
    ],
    "Streptomycin": Array [
      Array [
        "Resistance mutation found: K43R in gene rpsL",
        "Resistant allele coverage: 56",
        "Susceptible allele coverage: 0",
      ],
    ],
  },
  "inconclusive": Array [],
  "inducible": Array [],
  "resistant": Array [
    "Ofloxacin",
    "Moxifloxacin",
    "Isoniazid",
    "Kanamycin",
    "Ethambutol",
    "Streptomycin",
    "Ciprofloxacin",
    "Pyrazinamide",
    "Rifampicin",
    "Amikacin",
    "Capreomycin",
  ],
  "susceptibility": Object {
    "Amikacin": Object {
      "inconclusive": false,
      "inducible": false,
      "mutation": "rrs (A1401X)",
      "resistant": true,
      "susceptible": false,
    },
    "Capreomycin": Object {
      "inconclusive": false,
      "inducible": false,
      "mutation": "rrs (A1401X)",
      "resistant": true,
      "susceptible": false,
    },
    "Ciprofloxacin": Object {
      "inconclusive": false,
      "inducible": false,
      "mutation": "gyrA (D94G)",
      "resistant": true,
      "susceptible": false,
    },
    "Ethambutol": Object {
      "inconclusive": false,
      "inducible": false,
      "mutation": "embB (M306V)",
      "resistant": true,
      "susceptible": false,
    },
    "Isoniazid": Object {
      "inconclusive": false,
      "inducible": false,
      "mutation": "katG (S315X)",
      "resistant": true,
      "susceptible": false,
    },
    "Kanamycin": Object {
      "inconclusive": false,
      "inducible": false,
      "mutation": "rrs (A1401X)",
      "resistant": true,
      "susceptible": false,
    },
    "Moxifloxacin": Object {
      "inconclusive": false,
      "inducible": false,
      "mutation": "gyrA (D94G)",
      "resistant": true,
      "susceptible": false,
    },
    "Ofloxacin": Object {
      "inconclusive": false,
      "inducible": false,
      "mutation": "gyrA (D94G)",
      "resistant": true,
      "susceptible": false,
    },
    "Pyrazinamide": Object {
      "inconclusive": false,
      "inducible": false,
      "mutation": "pncA (H137R)",
      "resistant": true,
      "susceptible": false,
    },
    "Rifampicin": Object {
      "inconclusive": false,
      "inducible": false,
      "mutation": "rpoB (S450X)",
      "resistant": true,
      "susceptible": false,
    },
    "Streptomycin": Object {
      "inconclusive": false,
      "inducible": false,
      "mutation": "rpsL (K43R)",
      "resistant": true,
      "susceptible": false,
    },
  },
  "susceptible": Array [],
}
`;
